### PR TITLE
Add existing service

### LINKED ISSUE #238: Add existing content as item child - step 2: link services
## Motivations
When editing a chapter (should be easy adaptable to groups), allow choosing among existing items to add as child. 

## Subtasks
* Implement delayed search: when >3 char are typed and are not changed for 0.5sec (and only at that moment), launch the search to the search panel.
* Call the search service, with a limit of 4 results
* Display the loading spinner while searching is ongoing, error when occurs (without blocking everything)
* Make sure the previous searches are always cancelled when new requests are done
* When "+ Add" button is pressed, add the item at the end of the current children list.

### PR DESCRIPTION
**Pull request by AyeRlS**
Fixes #238
-
Step 2 of Add existing content as item child :
- Replace the static data with the actual data
- Add the selected item to the children of the current item.

### CODE REVIEW

File: src/app/modules/item/http-services/search-item.service.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  search(
+    searchString: string,
+    includedTypes?: ItemType[],
+    excludedTypes?: ItemType[],
+    limit = 4,
+  ): Observable<ItemFound[]> {
+    let params = new HttpParams().set('search', searchString);
+
+    if (limit) params = params.set('limit', limit.toString());
```
Comment (smadbe) [inline src/app/modules/item/http-services/search-item.service.ts:29]: `limit` is always set. With `if (limit) ` you are only filtering "0" but there is no reason not to put the limit if =0.

File: src/app/modules/shared-components/components/add-content/add-content.component.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angu
 import { FormBuilder, FormGroup } from '@angular/forms';
 import { Subscription } from 'rxjs';
 import { ItemType } from '../../../../shared/helpers/item-type';
+import { ChildData } from '../../../item/components/item-children-edit/item-children-edit.component';
```
Comment (smadbe) [inline src/app/modules/shared-components/components/add-content/add-content.component.ts:7]: You cannot make a component depends on a parent component.

File: src/app/modules/shared-components/components/add-content/add-content.component.ts [hunk 3]
```diff
@@ -25,33 +22,47 @@ export class AddContentComponent implements OnInit, OnDestroy {
...
+    this.subscriptions.push(
+      this.newContentForm.valueChanges.subscribe((changes: typeof defaultFormValues) => {
+        this.trimmedInputsValue = {
+          create: changes.create.trim(),
+          searchExisting: changes.searchExisting.trim(),
+        };
+      })
+    );
+
+    const existingTitleControl: Observable<string> | undefined = this.newContentForm.get('searchExisting')?.valueChanges;
```
Comment (smadbe) [inline src/app/modules/shared-components/components/add-content/add-content.component.ts:48]: "newContentForm" was not supposed to create new content at first? (vs addContent)

**Review by smadbe — CHANGES_REQUESTED** [review]
Looks good. Here are some comments. I'll think, I'll recheck the code in more details afterwards.

An item cannot be added twice as a child, so you should prevent that somehow. Either by greying out the "add" button in this case, or by failing properly on add (the list should not be reseted in this case).

File: src/app/modules/shared-components/components/add-content/add-content.component.ts (hunk 3 above)
Comment (AyeRlS) [reply src/app/modules/shared-components/components/add-content/add-content.component.ts]: I've renamed the form.

File: src/app/modules/item/http-services/search-item.service.ts (hunk 1 above)
Comment (AyeRlS) [reply src/app/modules/item/http-services/search-item.service.ts]: Fixed.

File: src/app/modules/shared-components/components/add-content/add-content.component.ts (hunk 2 above)
Comment (AyeRlS) [reply src/app/modules/shared-components/components/add-content/add-content.component.ts]: Fixed

**Review by smadbe — APPROVED** [review]
I rechecked into more details. Looks good 👍

**Pull request merged**